galaxy/views.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding:utf-8 -*
+#
+# Copyright 2023
+# - Skia <skia@hya.sk>
+#
+# Ce fichier fait partie du site de l'Association des Étudiants de l'UTBM,
+# http://ae.utbm.fr.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU General Public License a published by the Free Software
+# Foundation; either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Sofware Foundation, Inc., 59 Temple
+# Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+#
+
+from django.views.generic import DetailView, View
+from django.http import JsonResponse, Http404
+from django.db.models import Q, Case, F, When, Value
+from django.db.models.functions import Concat
+from django.utils.translation import gettext_lazy as _
+
+from core.views import (
+    CanViewMixin,
+    FormerSubscriberMixin,
+)
+from core.models import User
+from core.views import UserTabsMixin
+from galaxy.models import Galaxy, GalaxyLane
+
+
+class GalaxyUserView(CanViewMixin, UserTabsMixin, DetailView):
+    model = User
+    pk_url_kwarg = "user_id"
+    template_name = "galaxy/user.jinja"
+    current_tab = "galaxy"
+
+    def get_object(self, *args, **kwargs):
+        object = super(GalaxyUserView, self).get_object(*args, **kwargs)
+        try:
+            print(object.galaxy_user)
+        except User.galaxy_user.RelatedObjectDoesNotExist:
+            raise Http404(_("This citizen has not yet joined the galaxy"))
+        return object
```

Comment:
```suggestion
    def get_object(self, *args, **kwargs):
        user: User = super(GalaxyUserView, self).get_object(*args, **kwargs)
        if not hasattr(user, "galaxy_user"):
            raise Http404(_("This citizen has not yet joined the galaxy"))
        return user
```

Reply:
Oops, didn't finish the work here, sorry about that.